cypress/integration/rich-text/RichTextEditor.Tracking.spec.ts

```diff
@@ -0,0 +1,627 @@
+        ]);
+      });
+
+      it('adds column right', () => {
+        doAction('Add column right');
+
+        richText.expectTrackingValue([
+          insertTableAction(),
+          insertTableColumnAction({
+            tableSize: {
+              numColumns: 2,
+              numRows: 2,
+            },
+          }),
+        ]);
+      });
+
+      it('deletes row', () => {
+        doAction('Delete row');
+
+        richText.expectTrackingValue([
+          insertTableAction(),
+          removeTableRowAction({
+            tableSize: {
+              numColumns: 2,
+              numRows: 2,
+            },
+          }),
+        ]);
+      });
+
+      it('deletes column', () => {
+        doAction('Delete column');
+
+        richText.expectTrackingValue([
+          insertTableAction(),
+          removeTableColumnAction({
+            tableSize: {
+              numColumns: 2,
+              numRows: 2,
+            },
+          }),
+        ]);
+      });
+
+      it('deletes table', () => {
+        doAction('Delete table');
+
+        richText.expectTrackingValue([
+          insertTableAction(),
+          removeTableAction({
+            tableSize: {
+              numColumns: 2,
+              numRows: 2,
+            },
+          }),
+        ]);
+      });
+    });
+  });
+
+  describe('Links', () => {
+    const openCreateModal = (origin) => [
+      'openCreateHyperlinkDialog',
+      {
+        origin,
+      },
+    ];
+
+    const openEditModal = () => [
+      'openEditHyperlinkDialog',
+      {
+        origin: 'viewport-interaction',
+      },
+    ];
+
+    const insertHyperlink = (origin) => insert(origin, { linkType: 'uri', nodeType: 'hyperlink' });
+    const insertAssetHyperlink = (origin) =>
+      insert(origin, { linkType: 'Asset', nodeType: 'asset-hyperlink' });
+    const insertEntryHyperlink = (origin) =>
+      insert(origin, { linkType: 'Entry', nodeType: 'entry-hyperlink' });
+
+    const editHyperlink = () =>
+      edit('viewport-interaction', { linkType: 'uri', nodeType: 'hyperlink' });
+    const editAssetHyperlink = () =>
+      edit('viewport-interaction', { linkType: 'Asset', nodeType: 'asset-hyperlink' });
+    const editEntryHyperlink = () =>
+      edit('viewport-interaction', { linkType: 'Entry', nodeType: 'entry-hyperlink' });
+
+    const unlink = (origin) => [
+      'unlinkHyperlinks',
+      {
+        origin,
+      },
+    ];
+
+    const methods: [string, string, () => void][] = [
+      [
+        'using the link toolbar button',
+        'toolbar-icon',
+        () => {
+          richText.toolbar.hyperlink.click();
+        },
+      ],
+      [
+        'using the link keyboard shortcut',
+        'shortcut',
+        () => {
+          richText.editor.type(`{${mod}}k`);
+        },
+      ],
+    ];
+
+    for (const [triggerMethod, origin, triggerLinkModal] of methods) {
+      describe(triggerMethod, () => {
+        it('tracks adds and removes hyperlinks', () => {
```

Comment:
I think we are missing a test that opens the hyperlink modal but cancels without adding a link